lib/rules/prefer-regex-literals.js

```diff
@@ -50,14 +59,27 @@ module.exports = {
             url: "https://eslint.org/docs/rules/prefer-regex-literals"
         },
 
-        schema: [],
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    disallowRedundantWrapping: {
+                        type: "boolean",
+                        default: false
+                    }
+                },
+                additionalProperties: false
+            }
+        ],
 
         messages: {
-            unexpectedRegExp: "Use a regular expression literal instead of the 'RegExp' constructor."
+            unexpectedRegExp: "Use a regular expression literal instead of the 'RegExp' constructor.",
+            unexpectedRedundantRegExp: "Regular expression literal is unnecessarily wrapped within a 'RegExp' constructor."
```

Comment:
Just a question - would it make sense to have a different message if there's the `flags` argument?

```js
new RegExp(/abc/, 'u');
``` 

"Regular expression literal is unnecessarily wrapped within a 'RegExp' constructor" might be mistakenly interpreted as: this is same as `/abc/`

Reply:
Yes, I think that would be helpful. Do you have a suggestion?
Maybe `"Regular expression literal and modifiers are unnecessarily wrapped within a 'RegExp' constructor"`?

Reply:
`new RegExp(/abc/, 'u');` looks to me as the author wasn't aware of `/abc/u` syntax, so perhaps a message that suggests the fix (e.g., `"Use regular expression literal with flags instead of the 'RegExp' constructor."`, or something like `"Append flags to the regular expression literal instead of using the 'RegExp' constructor."`)?

Reply:
Done